hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntityFetchJoinedImpl.java

```diff
@@ -121,6 +122,7 @@ protected Initializer buildEntityJoinedFetchInitializer(
 			LockMode lockMode,
 			NotFoundAction notFoundAction,
 			DomainResult<?> keyResult,
+			DomainResult<Object> rowIdResult,
```

Comment:
Quick question:
`keyResult` is defined as `DomainResult<?>` while `rowIdResult` is `DomainResult<Object>`.

Is there any specific reason for using `Object` instead of `?`?

Reply:
The only reason is that `AbstractEntityInitializer` accepted a `DomainResult<Object>` typed parameter for the `rowIdResult`, so I just used that type. In general, looks like the row-id result is always typed as `DomainResult<Object>`, see also e.g. `AbstractEntityResultGraphNode#getRowIdResult`.

We probably could use `DomainResult<?>` everywhere, but that would mean a (slight) change of APIs.

Reply:
I've asked because I find it a bit inconsistent, but no big deal

Reply:
I agree, while the row-id typing was implemented like this we can always change it to `?`. Let's see what others think of this change before merging this PR.